zerver/tests/test_auth_backends.py

```diff
@@ -3175,6 +3175,15 @@ def test_wrong_password(self) -> None:
                                   dict(password="wrong"))
         self.assert_json_error(result, "Your username or password is incorrect.", 400)
 
+    def test_invalid_subdomain(self) -> None:
+        user = self.example_user('hamlet')
```

Comment:
it'd be a bit cleaner to have something like
```
username = 'hamlet`
user = self.example_user(username)
```

and then in the client_post we can re-use the `username` variable rather than hard-coding the `'hamlet'` string in yet another place.